### PR TITLE
[ New Feature ] test_compiler 优化

### PR DESCRIPTION
**Pull request by JewelRoam**
## 引入 GPU 计时器:

集成了 KernelBench 评测机制，当在 CUDA 设备上进行测试时（--device cuda），脚本现已采用基于 torch.cuda.Event 的 time_execution_with_cuda_event 函数进行计时。

对于 CPU 测试，保留了原有的 naive_timer 机制，并为其增加了与 GPU 模式类似的逐次试验（Trial-by-trial）的详细输出，统一了两种模式下的日志风格。

## 增加多次计时功能:

将计时试验次数（--trials）添加为命令行参数（默认5），允许用户根据评测需求灵活配置，方便进行快速调试或深度分析。

## 增加结构化导出功能

使用命令行参数 --output-dir 将模型/编译器配置参数组合下测定的性能数据存储到特定json文件中，方便后续分析。

### CODE REVIEW

File: graph_net/paddle/test_compiler.py [hunk 1]
```diff
@@ -112,12 +112,15 @@ def test_single_model(args):
...
     # build_strategy.build_cinn_pass = True
+    compilation_start_time = time.time()
     compiled_model = paddle.jit.to_static(
         model_dy,
         input_spec=input_spec,
         build_strategy=build_strategy,
         full_graph=True,
     )
+    compilation_end_time = time.time()
+    compilation_duration = compilation_end_time - compilation_start_time
```
Comment (lixinqi) [inline graph_net/paddle/test_compiler.py]: 这是不对的。这里其实并没有编译。Paddle真正的编译时机是第一次运行。torch 的真正编译甚至是第 3 次运行。

File: graph_net/paddle/test_compiler.py [hunk 2]
```diff
@@ -112,12 +112,15 @@ def test_single_model(args):
...
     compiled_model.eval()
     for _ in range(args.warmup if args.warmup > 0 else 0):
         compiled_model(**input_dict)
```
Comment (lixinqi) [inline graph_net/paddle/test_compiler.py]: 总共统计 warmup+1 次时间，最后取最大值，这个最大值再减去后续正式运行的时长，就当做是编译时长。
第一次是 decorator 时间，其他都是 warmup 每次执行的时间。

File: graph_net/torch/eval.py [hunk 3]
```diff
@@ -0,0 +1,765 @@
+"""
```
Comment (lixinqi) [inline graph_net/torch/eval.py]: eval.py 这个文件名不合适，会让人误以为到 torch 的 eval mode。

File: graph_net/paddle/test_compiler.py (hunk 1 above)
Comment (JewelRoam) [reply graph_net/paddle/test_compiler.py]: 我先在这个PR里把paddle的编译时间计算去掉了，目前torch也没做，后面准备好这个功能再另外提PR。

File: graph_net/paddle/test_compiler.py (hunk 2 above)
Comment (JewelRoam) [reply graph_net/paddle/test_compiler.py]: 我先在这个PR里把paddle的编译时间计算去掉了，目前torch也没做，后面准备好这个功能再另外提PR。

File: graph_net/torch/performance_eval.py [hunk 4]
```diff
@@ -0,0 +1,765 @@
+"""
```
Comment (lixinqi) [inline graph_net/torch/performance_eval.py]: 这些文件的内容用到了吗？

**Review by lixinqi — APPROVED** [review]
